packages/inferno-router/__tests__/router.spec.jsx

```diff
@@ -262,6 +273,98 @@ describe('Router (jsx)', () => {
 				() => render(<RouterContext location={ null }/>, container)
 			).to.throw(TypeError);
 		});
+		it('should fail when `matched` is not provided', () => {
+			expect(
+				() => render(<RouterContext location={ '/' } matched={ null }/>, container)
+			).to.throw(TypeError);
+		});
+		it('should pass when `location` is provided', () => {
+			const url = '/';
+			const matched = <GoodComponent/>;
+			const actual = render(<RouterContext location={ url } matched={ matched }/>, container);
+			expect(actual.props.location).to.equal(url);
+			expect(actual.props.matched).to.equal(matched);
+		});
+		it('should pass when `location` is provided and has percent encoded value', () => {
+			const url = '/100%25';
+			const matched = <GoodComponent/>;
+			const actual = render(<RouterContext location={ url } matched={ matched }/>, container);
+			expect(actual.props.location).to.equal(url);
+			expect(actual.props.matched).to.equal(matched);
+		});
+	});
+	describe('#match', () => {
+		it('should find route when url has normal value', () => {
+			const url = '/search/foo?arg1=50%25';
+			const history = createMemoryHistory();
+			history.push(url);
+
+			const router = <Router history={ history }>
+				<Route path="/" component={ BadComponent }>
+					<Route path="search/:searchData" component={ GoodComponent }/>
+				</Route>
+			</Router>;
+			console.log('history.location', history.location);
+			const renderProps = match(router, history.location.pathname + history.location.search);
+
+			const actual = render(<RouterContext {...renderProps}/>, container);
+
+			expect(actual.props.location).to.equal('/');
+			expect(
+				actual.props.matched.props.history.location.pathname + actual.props.matched.props.history.location.search
+			).to.equal('/search/foo?arg1=50%25');
+			expect(actual.props.matched.props.params.searchData).to.equal('foo');
+			expect(actual.props.matched.props.params.arg1).to.equal('50%');
```

Comment:
Since `params` extracted from paths are decoded, I though it made sense for search params to be decoded too.